main.go

```diff
@@ -40,6 +40,11 @@ func main() {
 		log.Fatalf("failed to create app: %v", err)
 	}
 
+	err = app.SetupLogger()
```

Comment:
IMO this initialization happens too late in the application cycle - we might lose some important logs from earlier. My proposed solution is:
- Move `SetupLogger` to a separate `logger` package.
- Call it at the top of `main`.